defold/defold
# Issue 3775 collision sticking

Allows flexible update for box2d `velocityThreshold` through game.project configuration options. Implemented in issue #3775 .

### Scope of changes

* Changes has been introduced to box2D bundled code so that it relies on a variable rather than a #define. 
* The core engine will load custom `velocityThreshold` and update the respective variable in Box2D when creating the Context2D.
* The new property has been added to the editor. No code changes there. A new number property under 'physics' section.

### Automated testing

* Since `velocityThreshold` is set to 1.0 by default and we need to change it in order to test, a new _customizable_ property was added to test_gamesys.h:ProjectOptions. A `VelocityThreshold2DTest` class was also added to initialized to a different value.
* You'll notice some renaming in two resources under `gamesys/.../test/collision_object` directory. It seems to me that having a static base and two dynamic objects moving on it seems a generic scenario deserving fewer resource files. So, the body.* and base.* files serve that purpose. They are also used by the older test with sleep/wakeup object. All tests are green in gamesys module.
* The idea of the test implemented is described in https://github.com/defold/defold/issues/3775#issuecomment-928973501

### Manual testing

A tried various tests with marginal velocities for colliding objects while tweaking velocityThreshold in game.project from within the new editor. I also loaded an old project in the new editor to make sure nothing breaks and defaults work properly. I didn't try to load a new project in the old editor.
